src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -2,40 +2,38 @@
 
 import org.skypro.skyshop.product.Product;
 
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+
 public class ProductBasket {
-    private Product[] products;
+    private List<Product> products;
     int counter;
 
     public int getCounter() {
         return counter;
     }
 
     public ProductBasket() {
-        this.products = new Product[5];
+        this.products = new LinkedList<>();
     }
 
     public void addProduct(Product product) {
-        if (counter >= 5) {
-            System.out.println("Невозможно добавить продукт");
-        } else {
-            products[counter] = product;
-            counter++;
-        }
+           products.add(product);
     }
 
     public double countBasketCost() {
         double basketCost = 0;
-        for (int i = 0; i < counter; ++i) {
-            basketCost = +basketCost + products[i].getPrice();
+        for (Product basketProduct: products) {
+            basketCost = +basketCost + basketProduct.getPrice();
```

Comment:
зачем плюс в начале?

```suggestion
            basketCost = basketCost + basketProduct.getPrice();
```